benchmark/test/test_framework.py.in

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env python3
+import subprocess
+import difflib
+import json
+import typing
+import re
+import pathlib
+import sys
+
+sourcepath = pathlib.Path("@CMAKE_CURRENT_SOURCE_DIR@")
+binpath = pathlib.Path("@PROJECT_BINARY_DIR@")
+generate = False
+if len(sys.argv) > 2 and sys.argv[2] == "--generate":
+    generate = True
```

Comment:
Ok, I misinterpreted that, test_framework is only used as module. Initially I thought it might also serve as an entry point because of cli args handling here. In that case I think it is not needed to wrap it into `__name__ ==  ...`.  Since its purpose would be to allow it to be used as module and a script. 

In terms of design, I think it would be cleaner to handle the cli args at the actual entry point and then pass it to the test_framework module, but for our purpose I think it is OK.